### PR TITLE
Replace etcd operator

### PR DESCRIPTION
**Pull request by mrIncompetent**
**What this PR does / why we need it**:
Replaces etcd operator with a StatefulSet

**Release note**:
```release-note
etcd operator has been replaced with a `StatefulSet`
```

### CODE REVIEW

File: api/pkg/controller/cluster/pending.go [hunk 1]
```diff
@@ -412,7 +409,8 @@ func (cc *Controller) ensureServices(c *kubermaticv1.Cluster) error {
 			return fmt.Errorf("failed to build Service: %v", err)
 		}
 
-		if diff := deep.Equal(service, existing); diff == nil {
+		diff := deep.Equal(service, existing)
```
Comment (alvaroaleman) [inline api/pkg/controller/cluster/pending.go]: Why change the notation here?

File: api/pkg/resources/etcd/discoveryservice.go [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+import (
+	"github.com/kubermatic/kubermatic/api/pkg/resources"
+	"k8s.io/apimachinery/pkg/util/intstr"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// DiscoveryService returns a service for the etcd
```
Comment (alvaroaleman) [inline api/pkg/resources/etcd/discoveryservice.go]: Maybe reword to "[...]for the etcd peers to find each other"?

File: api/pkg/resources/etcd/statefulset.go [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+		Namespace:   data.Cluster.Status.NamespaceName,
+	}
+
+	buf := bytes.Buffer{}
+	if err := tpl.Execute(&buf, tplData); err != nil {
+		return nil, err
+	}
+
+	return []string{
+		"/bin/sh",
```
Comment (alvaroaleman) [inline api/pkg/resources/etcd/statefulset.go:208]: Please omit the shell and just use `$(VARNAME)` notation in the cmd

File: api/pkg/resources/test/fixtures/deployment-aws-1.8.0-apiserver.yaml [hunk 4]
```diff
@@ -109,7 +109,7 @@ spec:
         - --insecure-port
         - "8080"
         - --etcd-servers
-        - http://etcd-cluster-client:2379
+        - http://etcd-client:2379
```
Comment (alvaroaleman) [inline api/pkg/resources/test/fixtures/deployment-aws-1.8.0-apiserver.yaml:112]: Just renaming it is actually a very nice move here :)

File: api/pkg/resources/etcd/statefulset.go (hunk 3 above)
Comment (mrIncompetent) [reply api/pkg/resources/etcd/statefulset.go:208]: Decided against it for now as the restore command definitly needs the shell.
We can later remove it.

**Review by alvaroaleman — APPROVED** [review]
